src/crypto/key_generator/key_generator_impl.cpp

```diff
@@ -218,49 +218,45 @@ namespace libp2p::crypto {
   outcome::result<KeyPair> KeyGeneratorImpl::generateKeys(
       Key::Type key_type) const {
     switch (key_type) {
-      case Key::Type::RSA1024:
-        return generateRsa(common::RSAKeyType::RSA1024);
-      case Key::Type::RSA2048:
-        return generateRsa(common::RSAKeyType::RSA2048);
-      case Key::Type::RSA4096:
-        return generateRsa(common::RSAKeyType::RSA4096);
-      case Key::Type::ED25519:
+      case Key::Type::RSA:
+        // TODO(akvinikym) 01.10.19 PRE-314: implement
+        BOOST_ASSERT_MSG(false, "not implemented");
+      case Key::Type::Ed25519:
         return generateEd25519();
-      case Key::Type::SECP256K1:
+      case Key::Type::Secp256k1:
         return generateSecp256k1();
+      case Key::Type::ECDSA:
+        BOOST_ASSERT_MSG(false, "not implemented");
       default:
         return KeyGeneratorError::UNSUPPORTED_KEY_TYPE;
     }
   }
 
   outcome::result<KeyPair> KeyGeneratorImpl::generateRsa(
       common::RSAKeyType bits_option) const {
-    int bits = 0;
-    Key::Type key_type;
-    switch (bits_option) {
-      case common::RSAKeyType::RSA1024:
-        bits = 1024;
-        key_type = Key::Type::RSA1024;
-        break;
-      case common::RSAKeyType::RSA2048:
-        bits = 2048;
-        key_type = Key::Type::RSA2048;
-        break;
-      case common::RSAKeyType::RSA4096:
-        bits = 4096;
-        key_type = Key::Type::RSA4096;
-        break;
-    }
-
-    // normally unreachable
-    if (0 == bits) {
-      return KeyGeneratorError::INCORRECT_BITS_COUNT;
-    }
-
-    OUTCOME_TRY(keys, detail::generateRsaKeys(bits));
-
-    return KeyPair{{{key_type, std::move(keys.first)}},
-                   {{key_type, std::move(keys.second)}}};
+    BOOST_ASSERT_MSG(false, "not implemented");
+
+    //    int bits = 0;
```

Comment:
I think these comments are not supposed to be here.

Reply:
Well, I think they can be left, as they give hint, how to implement the function, when the time comes

Reply:
Then I think that it would be great if it were explicitly written here.